projects/demo/src/app/cds-theme-path-location-strategy.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { APP_BASE_HREF, LocationStrategy, PathLocationStrategy, PlatformLocation } from '@angular/common';
+import { Inject, Injectable, Optional, Provider } from '@angular/core';
+import { UrlSerializer } from '@angular/router';
+import { cdsThemeAttribute, getCdsThemeFromDom } from './cds-theme-select.component';
+
+@Injectable()
+export class CdsThemePathLocationStrategy extends PathLocationStrategy {
+  constructor(
+    platformLocation: PlatformLocation,
+    @Optional() @Inject(APP_BASE_HREF) baseHref: string,
+    private readonly urlSerializer: UrlSerializer
+  ) {
+    super(platformLocation, baseHref);
+  }
+
+  prepareExternalUrl(internal: string): string {
```

Comment:
Might be helpful to add a comment here

Reply:
Comment added. Let me know if that doesn't make sense. I also added `override` keyword for clarity/maintainability.

https://github.com/vmware-clarity/ng-clarity/compare/6feead3954e6b2c8267a626bca77dd1f7b35304d..ff6910f69c625849ac15b051166cb8e4775ca9dd

Reply:
LGTM, I suppose it's a very literal comment of what is happening, but I think it's helpful to have it called out that we want to keep the theme persisted in all of our URLs for convenience when sharing.

Reply:
Okay, I updated the commit. I moved the comment to the class as I feel that makes more sense. I think the `override` keyword would point the reader in the right direction for that method specifically.

Reply:
Thank you! I was fine with the original version too, btw, but I appreciate the clarification.